dcpandas/df_builder.py

```diff
@@ -0,0 +1,292 @@
+
+    stat_all = dc.get_stat_all(places, stat_vars)
+    for place, place_data in stat_all.items():
+        if not place_data:
+            continue
+        for stat_var, stat_var_data in place_data.items():
+            if not stat_var_data:
+                continue
+            for source_series in stat_var_data['sourceSeries']:
+                time_series = source_series['val']
+                # Create a hashable for Observation options.
+                obs_options = (('measurementMethod',
+                                source_series.get('measurementMethod')),
+                               ('observationPeriod',
+                                source_series.get('observationPeriod')),
+                               ('unit', source_series.get('unit')),
+                               ('scalingFactor',
+                                source_series.get('scalingFactor')))
+                if mode == kseries:
+                    res[obs_options].append(
+                        dict({'place': place}, **time_series))
+                elif mode == kcov:
+                    date = max(time_series)
+                    res[stat_var][obs_options].append({
+                        'place': place,
+                        'date': date,
+                        'val': time_series[date]
+                    })
+    if mode == kseries:
+        return dict(res)
+    elif mode == kcov:
+        return {k: dict(v) for k, v in res.items()}
+
+
+def _time_series_pd_input(places, stat_var):
+    """Returns a `list` of `dict` per element of `places` based on the `stat_var`.
+
+    Data Commons will pick a set of Observation options that covers the
+    maximum number of queried places. Among ties, Data Commons selects an option
+    set with the latest Observation.
+
+    Args:
+      places (`str` or `iterable` of `str`): The dcids of Places to query for.
+      stat_var (`str`): The dcid of the StatisticalVariable.
+    Returns:
+      A `list` of `dict`, one per element of `places`. Each `dict` consists of
+      the time series and place identifier.
+
+    Examples:
+      >>> _time_series_pd_input(["geoId/29", "geoId/33"], "Count_Person")
+          [
+            {'2020-03-07': 20, '2020-03-08': 40, 'place': 'geoId/29'},
+            {'2020-08-21': 428, '2020-08-22': 429, 'place': 'geoId/33'}
+          ]
+    """
+
+    rows_dict = _group_stat_all_by_obs_options(places, [stat_var], 'series')
+    most_geos = []
+    max_geos_so_far = 0
+    latest_date = []
+    latest_date_so_far = ''
+    for options, rows in rows_dict.items():
+        current_geos = len(rows)
+        if current_geos > max_geos_so_far:
+            max_geos_so_far = current_geos
+            most_geos = [options]
+            # Reset tiebreaker stats. Recompute after this if-else block.
+            latest_date = []
+            latest_date_so_far = ''
+        elif current_geos == max_geos_so_far:
+            most_geos.append(options)
```

Comment:
why are we appending options (which are source keys / hashes) to most_geos? consider updating the name of most_geos